spec/rubocop/cop/lint/literal_as_condition_spec.rb

```diff
@@ -2,49 +2,119 @@
 
 RSpec.describe RuboCop::Cop::Lint::LiteralAsCondition, :config do
   %w(1 2.0 [1] {} :sym :"#{a}").each do |lit|
-    it "registers an offense for literal #{lit} in if" do
+    it "registers an offense for truthy literal #{lit} in if" do
       expect_offense(<<~RUBY, lit: lit)
         if %{lit}
            ^{lit} Literal `#{lit}` appeared as a condition.
           top
         end
       RUBY
+
+      expect_correction(<<~RUBY)
+        top
+      RUBY
+    end
+
+    it "registers an offense for truthy literal #{lit} in modifier if" do
+      expect_offense(<<~RUBY, lit: lit)
+        top if %{lit}
+               ^{lit} Literal `#{lit}` appeared as a condition.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        top
+      RUBY
+    end
+
+    it "registers an offense for truthy literal #{lit} in ternary" do
+      expect_offense(<<~RUBY, lit: lit)
+        %{lit} ? top : bar
+        ^{lit} Literal `#{lit}` appeared as a condition.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        top
+      RUBY
+    end
+
+    it "registers an offense for truthy literal #{lit} in unless" do
+      expect_offense(<<~RUBY, lit: lit)
+        unless %{lit}
+               ^{lit} Literal `#{lit}` appeared as a condition.
+          top
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+
+      RUBY
     end
 
-    it "registers an offense for literal #{lit} in while" do
+    it "registers an offense for truthy literal #{lit} in modifier unless" do
+      expect_offense(<<~RUBY, lit: lit)
+        top unless %{lit}
+                   ^{lit} Literal `#{lit}` appeared as a condition.
+      RUBY
+
+      expect_correction(<<~RUBY)
+
+      RUBY
+    end
+
+    it "registers an offense for truthy literal #{lit} in while" do
       expect_offense(<<~RUBY, lit: lit)
         while %{lit}
               ^{lit} Literal `#{lit}` appeared as a condition.
           top
         end
       RUBY
+
+      expect_correction(<<~RUBY)
+        while true
+          top
+        end
+      RUBY
     end
 
-    it "registers an offense for literal #{lit} in post-loop while" do
+    it "registers an offense for truthy literal #{lit} in post-loop while" do
       expect_offense(<<~RUBY, lit: lit)
         begin
           top
-        end while(%{lit})
+        end while %{lit}
                   ^{lit} Literal `#{lit}` appeared as a condition.
       RUBY
+
+      expect_correction(<<~RUBY)
+        while true
+          top
+        end
+      RUBY
```

Comment:
Should the form of loop change here? I think that maybe should be the domain of a separate Style cop.

Reply:
Fair enough. Fixed.